lib/ndebug/system/consoleproxy.c

```diff
@@ -35,6 +36,8 @@
 void consoleproxy_print_callback(print_callback_t *cb,
                                  const char *str, size_t len)
 {
+    if (arch_in_int_handler()) return;
```

Comment:
@travisg To your previous point about calling `printf` from IRQ context, this should drop any data from IRQ context on the floor.